src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let obj = Object.assign({}, state);
+
+  for (const action of actions) {
+    obj = Object.assign({}, obj, action.extraData);
+
+    if (action.type === 'addProperties') {
+      res.push(obj);
```

Comment:
1. push after conditions
2. just use either Object.assign or spread operator here
```suggestion
      Object.assign(obj, extraData)
```